posthog/api/test/test_insight.py

```diff
@@ -1124,6 +1131,138 @@ def test_insight_refreshing(self, spy_update_insight_cache) -> None:
+                    "query": TrendsQuery(
+                        series=[
+                            EventsNode(
+                                event="$pageview",
+                                properties=[
+                                    EventPropertyFilter(key="another", value="never_return_this", operator="is_not")
+                                ],
+                            )
+                        ]
+                    ).model_dump(),
+                    "dashboards": [dashboard_id],
+                },
+            ).json()
+            self.assertNotIn("code", response)  # Watching out for an error code
+            self.assertEqual(response["last_refresh"], None)
+
+            response = self.client.get(f"/api/projects/{self.team.id}/insights/{response['id']}/?refresh=true").json()
+            self.assertNotIn("code", response)
+            self.assertEqual(spy_execute_hogql_query.call_count, 1)
+            self.assertEqual(response["result"][0]["data"], [0, 0, 0, 0, 0, 0, 2, 0])
+            self.assertEqual(response["last_refresh"], "2012-01-15T04:01:34Z")
+            self.assertEqual(response["last_modified_at"], "2012-01-15T04:01:34Z")
+
+        with freeze_time("2012-01-15T05:01:34.000Z"):
+            _create_event(team=self.team, event="$pageview", distinct_id="1")
+            response = self.client.get(f"/api/projects/{self.team.id}/insights/{response['id']}/?refresh=true").json()
+            self.assertNotIn("code", response)
+            self.assertEqual(spy_execute_hogql_query.call_count, 2)
+            self.assertEqual(response["result"][0]["data"], [0, 0, 0, 0, 0, 0, 2, 1])
+            self.assertEqual(response["last_refresh"], "2012-01-15T05:01:34Z")
+            self.assertEqual(response["last_modified_at"], "2012-01-15T04:01:34Z")  # did not change
+
+        with freeze_time("2012-01-16T05:01:34.000Z"):
+            # load it in the context of the dashboard, so has last 14 days as filter
+            response = self.client.get(
+                f"/api/projects/{self.team.id}/insights/{response['id']}/?refresh=true&from_dashboard={dashboard_id}"
+            ).json()
+            self.assertNotIn("code", response)
+            self.assertEqual(spy_execute_hogql_query.call_count, 3)
+            self.assertEqual(
+                response["result"][0]["data"],
+                [
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    2.0,
+                    1.0,
+                    0.0,
+                ],
+            )
+            self.assertEqual(response["last_refresh"], "2012-01-16T05:01:34Z")
+            self.assertEqual(response["last_modified_at"], "2012-01-15T04:01:34Z")  # did not change
```

Comment:
Wait, does that mean it didn't change from the non-dashboard one? Shouldn't it be different? Or it didn't change since creation? In the latter case we might adjust the case to make it clear we get a different not-change for dashboards 🙈